testng-core/src/main/java/org/testng/SuiteRunner.java

```diff
@@ -415,16 +415,12 @@ private void runSequentially() {
     }
   }
 
-  private final AutoCloseableLock suiteResultsLock = new AutoCloseableLock();
-
   private void runTest(TestRunner tr) {
     visualisers.forEach(tr::addListener);
     tr.run();
 
     ISuiteResult sr = new SuiteResult(xmlSuite, tr);
-    try (AutoCloseableLock ignore = suiteResultsLock.lock()) {
-      suiteResults.put(tr.getName(), sr);
-    }
+    suiteResults.put(tr.getName(), sr);
```

Comment:
Could you explain why you removed the synchronisation?

Reply:
I had added the explanation in the commit message. 

`suiteResults` is backed by a `Collections.synchronizedMap()`. So it does not need any additional synchronization

Reply:
Right. But replace it by the synchronisation and it will remove the need of the extra collection later.

Reply:
Why do we need synchronized here especially when all that users are going to do is iterate. 

What are the impacts you forsee because of adding a new collection here which aims at giving a copy as unmodifiable collection?

Reply:
What is better between synchronisation + collection copy, or synchronisation only?

Reply:
I would say collection copy because its cheaper than unwanted synchronized blocks which are not adding any value.

Reply:
The other option is to bring back the usage of `lock` when adding elements to the map (which will now be just a plain vanilla `LinkedHashMap`) and return back an unmodifiable collection (and do away with the copy as well).

How does that sound ?

Reply:
I have made this change. Please check

Reply:
> I would say collection copy because its cheaper than unwanted synchronized blocks which are not adding any value. 

The synchronized block are hidden in synchronizedCollection

Reply:
Irrespective of where it resides the cost to enter and exit a critical section still exists. Why use synchronized when its not necessary and when we know there are going to be only reads and no writes